src/Rectangle.js

```diff
@@ -0,0 +1,138 @@
+     * @param {number} top Top bound of the rectangle.
+     */
+    initFromBounds (left, right, bottom, top) {
+        this.left = left;
+        this.right = right;
+        this.bottom = bottom;
+        this.top = top;
+    }
+
+    /**
+     * Initialize a Rectangle to the minimum AABB around a set of points.
+     * @param {Array.<Array.<number>>} points Array of [x, y] points.
+     */
+    initFromPointsAABB (points) {
+        this.left = Infinity;
+        this.right = -Infinity;
+        this.top = -Infinity;
+        this.bottom = Infinity;
+        for (let i = 0; i < points.length; i++) {
+            let x = points[i][0];
+            let y = points[i][1];
+            if (x < this.left) {
+                this.left = x;
+            }
+            if (x > this.right) {
+                this.right = x;
+            }
+            if (y > this.top) {
+                this.top = y;
+            }
+            if (y < this.bottom) {
+                this.bottom = y;
+            }
+        }
+    }
+
+    /**
+     * Determine if this Rectangle intersects some other.
+     * Note that this is a comparison assuming the Rectangle was
+     * initialized with Scratch-space bounds or points.
+     * @param {!Rectangle} other Rectangle to check if intersecting.
+     * @return {Boolean} True if this Rectangle intersects other.
+     */
+    intersects (other) {
+        return (
+            this.left <= other.right &&
+            other.left <= this.right &&
+            this.top >= other.bottom &&
+            other.top >= this.bottom
+        );
+    }
+
+    /**
+     * Determine if this Rectangle fully contains some other.
+     * Note that this is a comparison assuming the Rectangle was
+     * initialized with Scratch-space bounds or points.
+     * @param {!Rectangle} other Rectangle to check if fully contained.
+     * @return {Boolean} True if this Rectangle fully contains other.
+     */
+    contains (other) {
+        return (
+            other.left > this.left &&
+            other.right < this.right &&
+            other.top < this.top &&
+            other.bottom > this.bottom
+        );
+    }
+
+    /**
+     * Clamp a Rectangle to bounds.
+     * @param {number} left Left clamp.
+     * @param {number} right Right clamp.
+     * @param {number} bottom Bottom clamp.
+     * @param {number} top Top clamp.
+     */
+    clamp (left, right, bottom, top) {
+        this.left = Math.max(this.left, left);
+        this.right = Math.min(this.right, right);
+        this.bottom = Math.max(this.bottom, bottom);
+        this.top = Math.min(this.top, top);
+        // Ensure rectangle coordinates in order.
+        this.left = Math.min(this.left, this.right);
+        this.right = Math.max(this.right, this.left);
+        this.bottom = Math.min(this.bottom, this.top);
+        this.top = Math.max(this.top, this.bottom);
+    }
+
+    /**
+     * Push out the Rectangle to integer bounds.
+     */
+    ceil() {
```

Comment:
Minor nitpick:
The name `ceil`, to me, doesn't correspond well with what this function means. I think I would expect `bounds.ceil` to `ceil` each component rather than expand the bounds. I can see the argument that expanding the bounds could "feel like" a `ceil` operation, though... is that what you were going for?
Anyway, how would you feel about calling this something else? `snap` or `snapToInt` come to my mind, but if you have other thoughts I'd love to hear them. I'm also fine with leaving it `ceil` if that one feels right to you.